plugins/bigquery/dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -28,14 +29,37 @@
       {% set build_sql = create_table_as(False, target_relation, sql) %}
   {% else %}
      {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
+     
+     {%- set dest_partition = none -%}
+     
+     {#-- if partitioned, get the range of partition values to be updated --#}
+     {% if partition_by %}
+         {%- call statement('pro_tmp', fetch_result = True) -%}
+            {{ create_table_as(True, tmp_relation, sql) }}
+         {%- endcall -%}
+
+         {% call statement('get_partition_range', fetch_result = True) %}
+            select min({{partition_by}}), max({{partition_by}}) from {{tmp_relation}}
+         {% endcall %}
+         
+         {% set partition_range = load_result('get_partition_range').data[0]|list %}
+         {% set partition_min, partition_max = partition_range[0]|string, partition_range[1]|string %}
+         
+         {%- set dest_partition = {
+            'name': partition_by|lower|replace('date(','')|replace(')',''), 
```

Comment:
this is.... certainly clever :)

Maybe this is a good opportunity to add #1755 to dbt? I don't think we're going to want to lowercase the column name itself, as that could cause the eventual merge statement to break!

Reply:
Based on the conversation in #1755 and #1997, I went ahead and included the full `re` module for now. (I liked @beckjake's list, but I wasn't sure how to include constants in a concise way.) I recognize that should perhaps be a separate PR, I'm happy to merge/rebase here accordingly.

Reply:
I think allowing everything in `re.__all__` is probably fine, having looked through the module.